### DISCUSSION TITLE
Briefcase configuration error:  Configuration for 'statmachgen' is incomplete (missing 'description', 'sources')

### DISCUSSION THREAD

**Discussion by hlovatt** [beeware/toga#3066] [Debugging help]
Hi,

I had a working app with all the code in `app.py` and the file layout, `toml`, etc. generated by `briefcase new`. I then split the all-in-one code in `app.py` into three files `app.py`, `input.py`, and `model.py`. I now get the following error when I run `briefcase dev`:

```
Briefcase configuration error: Configuration for 'statmachgen' is incomplete (missing 'description', 'sources')
```

Same result for `briefcase update` and `briefcase dev -r`.

However, the `toml` does have them:

```
# This project was generated with 0.3.20 using template: https://github.com/beeware/briefcase-template@v0.3.20
[tool.briefcase]
project_name = "Pythonic Finite State Machine Generator"
bundle = "au.com.rhjod"
version = "0.0.0"
url = "https://rhjod.com.au/statmachgen"
license.file = "LICENSE"
author = "Howard Lovatt"
author_email = "howard@rhjod.com.au"

[tool.briefcase.briefcase.statmachgen]
formal_name = "Pythonic Finite State Machine Generator"
description = "From a finite state diagram to python code."
long_description = """From a finite state diagram the app generates python code using the `statmach` library.
"""
sources = [
    "src/statmachgen",
]
test_sources = [
    "tests",
]

requires = [
    "pydantic",
    "pydantic_extra_types",
]
test_requires = [
    "pytest",
]

...
```

Anyone know what's wrong?

Thanks in advance for any help,

Howard.

**Comment by freakboy3742** [comment]
Looks like a copy/paste or search/replace typo to me - the line:
```
[tool.briefcase.briefcase.statmachgen]
```
should be:
```
[tool.briefcase.app.statmachgen]
```

Not sure how that happened, but I'm guessing it's related to the rename you mentioned.

**Comment by hlovatt** [comment]
Fantastic. That's the problem. I guess in my refactoring in PyCharm it got changed. Many thanks.
